examples/rutie_ruby_example/Rakefile

```diff
@@ -1,9 +1,15 @@
+require "rbconfig"
 require "bundler/gem_tasks"
 require "rake/testtask"
 
 desc 'Build Rust extension'
 task :build_lib do
-  sh 'cargo build --release'
+  case RbConfig::CONFIG['host_os']
+  when /darwin|mac os/
+    sh 'cargo rustc --release -- -C link-args=-Wl,-undefined,dynamic_lookup'
```

Comment:
The example wasn't working either on mac, so I fixed that as well

Reply:
I think this is good for the most part.  But if you're doing static linking in this PR doesn't `dynamic_lookup` mean the opposite in this case?

Reply:
_Not that I'm against dynamic lookup as dynamic linking is recommended._

Reply:
Wouldn't the example still create a `dylib`? 🤔